public/app/plugins/datasource/cloudwatch/components/LogsQueryField.tsx

```diff
@@ -33,59 +32,36 @@ export interface CloudWatchLogsQueryFieldProps
     Themeable2 {
   absoluteRange: AbsoluteTimeRange;
   onLabelsRefresh?: () => void;
-  ExtraFieldElement?: ReactNode;
+  ExtraFieldElement?: React.ReactNode;
   exploreId: ExploreId;
   query: CloudWatchLogsQuery;
 }
 
-const addPaddingToButton = css`
-  padding: 1px 4px;
-`;
-interface State {
-  hint:
-    | {
-        message: string;
-        fix: {
-          label: string;
-          action: () => void;
-        };
-      }
-    | undefined;
-}
-
-class CloudWatchLogsQueryField extends React.PureComponent<CloudWatchLogsQueryFieldProps, State> {
-  state: State = {
-    hint: undefined,
-  };
-
-  plugins: Array<Plugin<Editor>>;
+export const CloudWatchLogsQueryField = (props: CloudWatchLogsQueryFieldProps) => {
+  const { query, datasource, onChange, onRunQuery, ExtraFieldElement, data } = props;
 
-  constructor(props: CloudWatchLogsQueryFieldProps, context: React.Context<any>) {
-    super(props, context);
+  const showError = data && data.error && data.error.refId === query.refId;
+  const cleanText = datasource.languageProvider ? datasource.languageProvider.cleanText : undefined;
 
-    this.plugins = [
-      BracesPlugin(),
-      SlatePrism(
-        {
-          onlyIn: (node: Node) => node.object === 'block' && node.type === 'code_block',
-          getSyntax: (node: Node) => 'cloudwatch',
-        },
-        { ...(prismLanguages as LanguageMap), cloudwatch: syntax }
-      ),
-    ];
-  }
-
-  componentDidMount = () => {
-    const { query, datasource, onChange } = this.props;
-
-    if (onChange) {
-      onChange({ ...query, logGroupNames: query.logGroupNames ?? datasource.logsQueryRunner.defaultLogGroups });
+  const plugins: Array<Plugin<Editor>> = [
+    BracesPlugin(),
+    SlatePrism(
+      {
+        onlyIn: (node: Node) => node.object === 'block' && node.type === 'code_block',
+        getSyntax: (node: Node) => 'cloudwatch',
+      },
+      { ...(prismLanguages as LanguageMap), cloudwatch: syntax }
+    ),
+  ];
+
+  useEffectOnce(() => {
+    if (!query.logGroupNames) {
+      onChange({ ...query, logGroupNames: datasource.logsQueryRunner.defaultLogGroups });
```

Comment:
@idastambuk is working on setting a [default query](https://github.com/grafana/cloud-data-sources/issues/79) in the data source. The default query will be applied before the query editor components are rendered the first time. This issue has been blocked because there's been a bug in Grafana core's handling of default queries, but I just merged a [fix](https://github.com/grafana/grafana/pull/59625) related to that so this issue should now be unblocked. 

Anyway, we should be able to set logGroupNames to equal the defaultLogGroups in the default query setter rather than calling `onChange` when rendering this component. Maybe @fridgepoet and @idastambuk can coordinate on this so that Ida's PR gets merged first, and then we can remove this effect before merging this PR?

Reply:
Default query feature merged in https://github.com/grafana/grafana/pull/60221 🙌🏻